File: src/core/updateProfile.php
<?php
	require_once 'config.php';
	$post = $_POST;
	var_dump($post);
	var_dump($_FILES["avatar"]);
	$user_id = $post['user_id'];
	$mode = $post['mode'];
	$value = $post['value'];
	switch ($mode) {
	case 'name':
		queryn('UPDATE user SET name = :name WHERE user_id = :user_id',array('user_id'=>$user_id, 'name' => $value));
		break;
	case 'tanggal_lahir':
		queryn('UPDATE user SET birthdate = :birthdate WHERE user_id = :user_id',array('user_id'=>$user_id, 'birthdate' => $value));
		break;
	case 'password':
		queryn('UPDATE user SET password = :password WHERE user_id = :user_id',array('user_id'=>$user_id, 'password' => $value));
		break;
	case 'avatar':
		move_uploaded_file($_FILES["avatar"]["tmp_name"],"avatar/" . $user_id . '.jpg');
	}	
?>

File: IF3038-2013/src/java/org/progin/todo/UpdateProfile.java
package org.progin.todo;
import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.json.JSONArray;

/**
 * Handler for core/updateProfile
 *
 * @author Irfan Kamil
 * @author Hanif Eridaputra
 * @author Damiann
 */
public class UpdateProfile extends HttpServlet {

    private static final long serialVersionUID = 1L;

    @Override
    public void doGet(HttpServletRequest request,
            HttpServletResponse response)
            throws IOException, ServletException {
        String user_id = request.getParameter("user_id");
        String value = request.getParameter("value");
        JSONArray json;
        String mode = request.getParameter("mode");
        String[] param = {value,user_id};
        if (mode.compareTo("name") == 0) {
            Query.n("update user SET name = ? WHERE user_id = ?", param);
        } else if (mode.compareTo("tanggal_lahir") == 0) {
            Query.n("update user SET birthdate = ? WHERE user_id = ?", param);
        } else if (mode.compareTo("password") == 0) {
            Query.n("update user SET password = ? WHERE user_id = ?", param);
        }

    }

    @Override
    public void doPost(HttpServletRequest request,
            HttpServletResponse response)
            throws IOException, ServletException {
        doGet(request, response);
    }
}
